infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -50,10 +50,23 @@ public BaseServer(@Nonnull ServerContext context) {
      * @param r     The server router.
      */
     @ServerHandler(type = CorfuMsgType.PING)
-    private static void ping(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    private void ping(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         r.sendResponse(ctx, msg, CorfuMsgType.PONG.msg());
     }
 
+    /**
+     * Respond to a keep alive message.
+     * Note: this message ignores epoch.
+     *
+     * @param msg   The incoming message
+     * @param ctx   The channel context
+     * @param r     The server router.
+     */
+    @ServerHandler(type = CorfuMsgType.KEEP_ALIVE)
```

Comment:
Keep alive seems more descriptive since this is the purpose of this message. Also we already have HEARTBEAT_REQUEST/HEARTBEAT_RESPONSE, use HEART_BEAT again brings confusion.

Reply:
Why do you need to introduce a new message? Why not just use ping ?

Reply:
@Maithem Ping is epoch aware, and being used getOrchestrator() and getClusterStatus(). This new message ignores epoch since keep alive doesn't care epoch. Otherwise using ping for keep alive could result in WrongEpochException.